src/providers/base_javascript.js

```diff
@@ -1,212 +1,339 @@
 				}
 			});
 	}
 
+	/**
+   * Creates a SBOM containing only direct dependencies
+   * @param {Object} [opts={}] - Optional configuration options
+   * @returns {string} The SBOM as a JSON string
+   * @private
+   */
+	#getDirectDependencySbom(opts = {}) {
+		const depTree = this._buildDependencyTree(false);
+		let mainComponent = toPurl(ecosystem, this.#manifest.name, this.#manifest.version);
+
+		let sbom = new Sbom();
+		sbom.addRoot(mainComponent);
+
+		const rootDeps = this._getRootDependencies(depTree);
+		const sortedDepsKeys = Array
+			.from(rootDeps.keys())
+			.filter(key => this.#manifest.dependencies.includes(key))
+			.sort();
+		for (const key of sortedDepsKeys) {
+			const rootPurl = toPurlFromString(sbom.getRoot().purl);
+			sbom.addDependency(rootPurl, rootDeps.get(key));
+		}
+		sbom.filterIgnoredDeps(this.#manifest.ignored);
+		return sbom.getAsJsonString(opts);
+	}
+
+	/**
+   * Extracts root dependencies from the dependency tree
+   * @param {Object} depTree - The dependency tree object
+   * @returns {Map<string, PackageURL>} Map of dependency names to their PackageURL objects
+   * @protected
+   */
+	_getRootDependencies(depTree) {
+		if (!depTree.dependencies) {
+			return new Map();
+		}
+
+		return new Map(
+			Object.entries(depTree.dependencies).map(
+				([key, value]) => [key, toPurl(ecosystem, key, value.version)]
+			)
+		);
+	}
+
+	/**
+   * Executes the list command to get dependencies
+   * @param {boolean} includeTransitive - Whether to include transitive dependencies
+   * @param {string} manifestDir - The manifest directory
+   * @returns {string} The command output
+   * @private
+   */
 	#executeListCmd(includeTransitive, manifestDir) {
 		const listArgs = this._listCmdArgs(includeTransitive, manifestDir);
 		return this.#invokeCommand(listArgs);
 	}
 
-	#version() {
-		this.#invokeCommand(['--version'], { stdio: 'ignore' });
+	/**
+   * Gets the version of the package manager
+   * @returns {string} The version string of the package manager
+   * @protected
+   */
+	_version() {
+		return this.#invokeCommand(['--version']);
 	}
 
+	/**
+   * Creates or updates the lock file for the package manager
+   * @param {string} manifestDir - Directory containing the manifest file
+   * @private
+   */
 	#createLockFile(manifestDir) {
-		// in windows os, --prefix flag doesn't work, it behaves really weird , instead of installing the package.json fromm the prefix folder,
-		// it's installing package.json (placed in current working directory of process) into prefix directory, so
-		let originalDir = process.cwd()
-		if (os.platform() === 'win32') {
-			process.chdir(manifestDir)
+		const originalDir = process.cwd();
+		const isWindows = os.platform() === 'win32';
+
+		if (isWindows) {
+			// On Windows, --prefix flag doesn't work as expected
+			// Instead of installing from the prefix folder, it installs from current working directory
+			process.chdir(manifestDir);
```

Comment:
I wonder can we just set cwd in opts here as well, or do we really have to use chdir?

Reply:
I was considering doing the same for all commands and pkgManagers but I will need to verify on win/mac/linux on a different PR, maybe?

Reply:
Ill do some testing on windows once this PR is merged then :+1:

Reply:
Tested removing this on windows and it appears to work fine with `cwd` being set and `process.chdir` removed